docs/rfcs/rfc-018-mathematical-operators.md

```diff
@@ -0,0 +1,100 @@
+# Mathematical operators
+
+## Problem
+As described in the issue #677, if we have to do a mathematical operation using the GraphQL library, 
+the common approach is to query the database and then perform the mutation with the final property value.
+The former approach has several downsides:
+- It requires multiple roundtrips to the database.
+- It's not concurrency safe.
+- It requires more code than necessary in the client.
+
+## Proposed Solution
+Add new input properties that solve atomic operations like:
+- `_INC`
+- `_DEC`
+- `_MULT`
+- `_DIV`
+
+For instance `counter_INC` or `counter_DEC`.
+These properties should be available for all the following types:
+- Int
+- Float
+- BigInt
+
+### Usage Examples
+```graphql
+updateNodes(
+  where: { id: "e9bc687a-efd1-419d-b208" }
+  update: {
+    counter_INC: 1,
+  }
+)
+```
+
+### Technical considerations
+#### Ambiguous property
+Given the following example: 
+```graphql
+updateNodes(
+  where: { id: "e9bc687a-efd1-419d-b208" }
+  update: {
+    counter: 10,
+    counter_INC: 1,
+  }
+)
+```
+`counter` is ambiguous, and a strategy for this situation should be defined.
+
+#### Schema size
+The proposed solution could increase the size of the augmented schema noticeably.
+
+## Alternative Solution
+Create a new GraphQL scalar type that supports these operations:
+- `INC`
+- `DEC`
+- `MULT`
+- `DIV`
+
+For instance `counter: { INC: 1 }` or `counter: { DEC: 1 }`, the syntax `counter: 1` should remains valid.
+The new GraphQL scalar type should be available for all the following types:
+- Int
+- Float
+- BigInt
+
+### Usage Examples
+```graphql
+updateNodes(
+  where: { id: "e9bc687a-efd1-419d-b208" }
+  update: {
+    counter: { inc: 1 },
+  }
+)
+```
+### Technical considerations
+#### Inconsistent design
+The proposed solution implements a design different from similar features like [https://neo4j.com/docs/graphql/manual/current/filtering/](https://neo4j.com/docs/graphql/manual/current/filtering/),
+and increase the learning curve of the library.
+
```

Comment:
Another consideration to keep in mind, is the feasibility of having a scalar and object union type.
i.e., the same input property to support both `counter: 5` and `counter: { INC: 1}`

Reply:
I Agree with that, the whole solution stands on the assumption that the workaround of creating a custom scalar type to be used as a wrapper in the augmented schema is achievable. I add this as a consideration to improve the visibility.

Reply:
Please, let me know if this could be solved by 99701e156480bd92348ed987ab2f12df7611ba18 @angrykoala

Reply:
@darrellwarde knows more about graphql limitations, but I think this is not doable, at least not with quite a good bit of hacks